src/main/java/uk/gov/companieshouse/company/profile/service/CompanyProfileService.java

```diff
@@ -383,5 +383,29 @@ public void deleteCompanyProfile(String companyNumber) throws ResourceNotFoundEx
 
     }
 
+    /** Set can_file based on company type and status. */
+    public void determineCanFile(String companyNumber) {
+        try {
+            CompanyProfileDocument companyProfileDocument =
+                    getCompanyProfileDocument(companyNumber);
+            String companyType = companyProfileDocument.getCompanyProfile().getType();
+            String companyStatus = companyProfileDocument.getCompanyProfile().getCompanyStatus();
+
+            if (companyType.equals("ltd")
+                    || companyType.equals("llp")
+                    || companyType.equals("plc")
+                    || companyType.contains("private")) {
```

Comment:
Do we have the full list of company types? Are we sure none of them contain 'private' other than the ones we are checking for?

Reply:
"private-unlimited", "private-limited-guarant-nsc-limited-exemption", "private-limited-guarant-nsc", "private-unlimited-nsc", "private-limited-shares-section-30-exemption" are the only enums that contain "private" and they are all mentioned in the ticket. line 397 should still be able to recognise them all